plugins/BEdita/Core/src/Model/Table/UsersTable.php

```diff
@@ -295,6 +302,73 @@ public function beforeDelete(Event $event, EntityInterface $entity)
         }
     }
 
+    /**
+     * Modify user entity to become anonymous and hidden
+     *
+     * @param \Cake\Datasource\EntityInterface $entity the entity to anonimize
+     * @return void
+     */
+    protected function anonymizeUser(EntityInterface $entity)
+    {
+        $notNull = $this->notNullableColumns($this);
+        foreach ($this->inheritedTables() as $table) {
+            $notNull = array_merge($notNull, $this->notNullableColumns($table));
+        }
+        $properties = array_diff((array)$entity->visibleProperties(), $notNull, ['type']);
+        foreach ($properties as $name) {
+            $entity->set($name, null);
+        }
+        $deletedValue = self::DELETED_USER_PREFIX . $entity->get('id');
+        $entity->set('username', $deletedValue);
+        $entity->set('uname', $deletedValue);
+        $entity->set('password', null);
+        $entity->set('deleted', true);
+        $entity->set('locked', true);
+        $this->saveOrFail($entity);
+    }
+
+    /**
+     * Retrieve not nullable columns for a table
+     *
+     * @param \Cake\ORM\Table $table Table class
+     * @return array
+     */
+    protected function notNullableColumns($table)
```

Comment:
[minor] This method may be static. Since it's a helper function, I personally would feel it would be slightly more explicit if it was `protected static`, but of course this is a matter of personal taste.